streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -109,16 +109,27 @@ public StateDirectory(final StreamsConfig config, final Time time, final boolean
                 log.warn("Using /tmp directory in the state.dir property can cause failures with writing the checkpoint file" +
                     " due to the fact that this directory can be cleared by the OS");
             }
-
             // change the dir permission to "rwxr-x---" to avoid world readable
-            final Path basePath = Paths.get(baseDir.getPath());
-            final Path statePath = Paths.get(stateDir.getPath());
+            configurePermissions(Paths.get(baseDir.getPath()));
+            configurePermissions(Paths.get(stateDir.getPath()));
+        }
+    }
+    
+    private void configurePermissions(final Path path) {
+        if (path.getFileSystem().supportedFileAttributeViews().contains("posix")) {
             final Set<PosixFilePermission> perms = PosixFilePermissions.fromString("rwxr-x---");
             try {
-                Files.setPosixFilePermissions(basePath, perms);
-                Files.setPosixFilePermissions(statePath, perms);
+                Files.setPosixFilePermissions(path, perms);
             } catch (final IOException e) {
-                log.error("Error changing permissions for the state or base directory {} ", stateDir.getPath(), e);
+                log.error("Error changing permissions for the directory {} ", path, e);
+            }
+        } else {
+            final File file = path.toFile();
+            boolean set = file.setReadable(true, false);
```

Comment:
I think we actually want it to be readable _only_ by the user, and explicitly restrict permissions for all other users. The patch which originally broke things for Windows users was trying to tighten up the security in exactly this way

Reply:
There's an imperfect mapping between POSIX-style permissions and what can be configured using the File API as it's just owner and everyone else. I'll change it so it's only readable by the owner.

Reply:
For symmetry, I'll do the same for `setExecutable` as well.